src/monstergenerator.cpp

```diff
@@ -328,6 +328,14 @@ void load_monster_adjustment( const JsonObject &jsobj )
     adjustments.push_back( adj );
 }
 
+static void build_behavior_tree( mtype &type )
+{
+    type.set_strategy();
+    if( type.has_flag( MF_ABSORBS ) || type.has_flag( MF_ABSORBS_SPLITS ) ) {
```

Comment:
I guess we will replace flags with goals in monster definition later (or maybe add monster flags to goals definitions)?

Reply:
This should probably get unified around special attacks, or maybe rename them to "abilities", but I haven't thought about that too much yet.
The behavior elements could even be inlined along with metadata about the attack itself.